keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,491 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently the most container runtimes work is by masking and setting as read-only certain paths in `/proc`.
+This is to prevent data from being exposed into a container that should not be.
+However, there are certain use-cases where it is necessary to turn this off.
+
+In 1.12, this was introduced as the ProcMountType feature gate, and has languished since. This KEP is
+a successor to (and heavily based on) https://github.com/kubernetes/community/pull/1934/, updated for the modern era.
+
+## Motivation
+
+For end-users who would like to run unprivileged containers using user namespaces _nested inside_ CRI containers, we need an option to have a `ProcMount`.
+That is, we need an option to designate explicitly turn off masking and setting read-only of paths so that we can mount `/proc` in the nested container as an unprivileged user.
+
+Please see the following filed issues for more information:
+- [opencontainers/runc#1658](https://github.com/opencontainers/runc/issues/1658#issuecomment-373122073)
+- [moby/moby#36597](https://github.com/moby/moby/issues/36597)
+- [moby/moby#36644](https://github.com/moby/moby/pull/36644)
+
+### Goals
+
+- Add the ProcMountType option to the pod API, allowing users to override the proc mount behavior in a container.
+
+### Non-Goals
+
+## Proposal
+
+
+Add a new `string` type field named `ProcMountType` will hold the viable
+options for `procMount` to the `SecurityContext`
+definition.
+
+By default,`procMount` is `default`, aka the same behavior as today and the
+paths are masked.
+
+This will look like the following in the spec:
+
+```go
+type ProcMountType string
+
+const (
+    // DefaultProcMount uses the container runtime default ProcType.  Most 
+    // container runtimes mask certain paths in /proc to avoid accidental security
+    // exposure of special devices or information.
+    DefaultProcMount ProcMountType = "Default"
+
+    // UnmaskedProcMount bypasses the default masking behavior of the container
+    // runtime and ensures the newly created /proc the container stays in tact with
+    // no modifications.  
+    UnmaskedProcMount ProcMountType = "Unmasked"
+)
+
+procMount *ProcMountType
```

Comment:
`nil` is default?

Also current documentaiton is saying: "// Note that this field cannot be set when spec.os.name is windows."

Can we update this by specifying what to expect when it is Windows. Which component will blow out and what is expected Runtime reaction on this

Reply:
this is nit for the implementation. Also would be good to add a test for Windows setting this value

Reply:
agreed